variables.tf

```diff
@@ -98,8 +102,11 @@ variable "resource_group_name" {
 
 variable "kubernetes_version" {
   type        = string
-  default     = "1.26"
-  description = "Specify which Kubernetes release to use. The default used is the latest Kubernetes version available in the region"
+  default     = "1.28"
```

Comment:
Was this version bump required to support the UDR functionality?

If not, I would suggest we come up with a process for bumping default versions like this. There could be a scenario in the future where someone hasn't supplied this value and I'm not sure what the expected behavior is in that case. Would they destroy/create the cluster or would there be an update in place if they consumed this new feature? If it's a destroy/create it seems like we'd want to reconsider this as a feature release and potentially classify it as breaking.

A better long-term solution might be to require the user to supply this value rather than us trying to default it for them.